### PR TITLE
Moving NodeInfo.getVersion external usages to String

### PR DESCRIPTION
**Pull request by ldematte**
First (simple) step for removing Version from NodeInfo.
Move external users to use a String version; this should be safe as NodeInfo.getVersion was used just for serialization purposes.

### CODE REVIEW

**Comment by ldematte** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1

**Comment by ldematte** [comment]
@elasticsearchmachine rerun elasticsearch-ci/bwc elasticsearch-ci/packaging-tests-unix-sample elasticsearch-ci/part-3

File: x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDocTests.java [hunk 1]
```diff
@@ -313,7 +313,7 @@ public void testToXContent() throws IOException {
         final List<XPackFeatureSet.Usage> usageList = singletonList(new MonitoringFeatureSetUsage(false, null));
 
         final NodeInfo mockNodeInfo = mock(NodeInfo.class);
-        Version mockNodeVersion = Version.CURRENT.minimumCompatibilityVersion();
+        var mockNodeVersion = Build.current().version();
```
Comment (rjernst) [inline x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDocTests.java:316]: Shouldn't this be `Build.current().minWireCompatVersion()`?

Comment (rjernst) [reply x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDocTests.java:316]: (also appears unrelated to the PR title?)

Comment (ldematte) [reply x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDocTests.java:316]: It should not matter, it is just a placeholder for a version, but you are right, better not introducing any difference

Comment (ldematte) [reply x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/collector/cluster/ClusterStatsMonitoringDocTests.java:316]: > (also appears unrelated to the PR title?)

Well, I need to change the mocked version from `Version` to `String`, it seemed silly to change it to `Version.CURRENT.minimumCompatibilityVersion().toString()`. 
I can replace it with any string - I will replace it with `randomAlphaOfLengthBetween(6, 32)` to make it more explicit that it is "any string you can use for a version".

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**